### PR TITLE
Update PTCA image to address torch vulnerabilities

### PR DESCRIPTION
**Pull request by raviskolli**

### CODE REVIEW

File: assets/training/automl/environments/ai-ml-automl-dnn-text-gpu-ptca/context/Dockerfile [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
 
 USER root:root
 
 RUN pip install 'azureml-automl-dnn-nlp=={{latest-pypi-version}}'
 RUN pip install 'azureml-defaults=={{latest-pypi-version}}'
 
-RUN pip install torch==1.13.1
+RUN pip uninstall -y onnxruntime
+
+RUN pip install onnx==1.16.2
```
Comment (jeff-shepherd) [inline assets/training/automl/environments/ai-ml-automl-dnn-text-gpu-ptca/context/Dockerfile:10]: If pnnx==1.16.2 is important, we should probably update the pins in  azureml-automl-runtime, azureml-train-automl-runtime and azureml-training-tabular, which currently have 1.16.1

File: assets/training/automl/environments/ai-ml-automl-dnn-text-gpu-ptca/context/Dockerfile [hunk 2]
```diff
@@ -1,20 +1,24 @@
...
-
-RUN pip install transformers==4.36.2
-RUN pip install optimum==1.16.1
-RUN pip install accelerate==0.26.1
-RUN pip install deepspeed==0.13.1
+RUN pip install -i https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/onnxruntime-cuda-12/pypi/simple/ onnxruntime-training==1.18.0
+RUN TORCH_CUDA_ARCH_LIST="5.2;6.0;7.0;8.0;8.6;9.0" python -m onnxruntime.training.ortmodule.torch_cpp_extensions.install
+RUN pip install torch-ort==1.18.0 && TORCH_CUDA_ARCH_LIST="5.2;6.0;7.0;8.0;8.6;9.0" python -m torch_ort.configure
+
+RUN pip uninstall -y onnxruntime
```
Comment (jeff-shepherd) [inline assets/training/automl/environments/ai-ml-automl-dnn-text-gpu-ptca/context/Dockerfile:16]: Why uninstall onnxruntime (twice) ?
It is a dependency of azureml-automl-runtime, azureml-train-automl-runtime and azureml-training-tabular

File: assets/training/automl/environments/ai-ml-automl-dnn-text-gpu-ptca/context/Dockerfile (hunk 1 above)
Comment (raviskolli) [reply assets/training/automl/environments/ai-ml-automl-dnn-text-gpu-ptca/context/Dockerfile:10]: The other packages don't need this update. This is specific to onnxruntime-training which is just used in PTCA environment.

File: assets/training/automl/environments/ai-ml-automl-dnn-text-gpu-ptca/context/Dockerfile (hunk 2 above)
Comment (raviskolli) [reply assets/training/automl/environments/ai-ml-automl-dnn-text-gpu-ptca/context/Dockerfile:16]: During training, some namespace dependency resolution causes conflicts when both onnxruntime and onnxruntime-training are installed. This is just to safeguard that it's not present after all the package installations.

**Pull request merged**